replica/table.cc

```diff
@@ -1350,59 +1351,82 @@ future<> table::snapshot(database& db, sstring name) {
     tlogger.debug("snapshot {}", jsondir);
 
     auto sstable_deletion_guard = co_await get_units(_sstable_deletion_sem, 1);
+    std::exception_ptr ex;
 
-        // FIXME: indentation
-        auto tables = boost::copy_range<std::vector<sstables::shared_sstable>>(*_sstables->all());
+    std::vector<sstables::shared_sstable> tables;
+    try {
+        tables = boost::copy_range<std::vector<sstables::shared_sstable>>(*_sstables->all());
         co_await io_check([&jsondir] { return recursive_touch_directory(jsondir); });
         co_await max_concurrent_for_each(tables, db.get_config().initial_sstable_loading_concurrency(), [&db, &jsondir] (sstables::shared_sstable sstable) {
-                  return with_semaphore(db.get_sharded_sst_dir_semaphore().local(), 1, [&jsondir, sstable] {
-                    return io_check([sstable, &dir = jsondir] {
-                        return sstable->create_links(dir);
-                    });
-                  });
+            return with_semaphore(db.get_sharded_sst_dir_semaphore().local(), 1, [&jsondir, sstable] {
+                return io_check([sstable, &dir = jsondir] {
+                    return sstable->create_links(dir);
                 });
+            });
+        });
         co_await io_check(sync_directory, jsondir);
+    } catch (...) {
```

Comment:
Maybe this answers my question on the previous patch - it seems you are adding a "catch" where we previously had finally. So perhaps we should merge these two patches?

Reply:
Maybe merging the patches would indeed make the overall change clearer and easier to review.